centreon/www/class/centreonGraph.class.php

```diff
@@ -456,16 +469,20 @@ public function initCurveList()
 
         /* Manage reals metrics */
         if (isset($l_rselector)) {
-            $DBRESULT = $this->DBC->query(
+            $DBRESULT = $this->DBC->prepare(
                 "SELECT host_id, service_id, metric_id, metric_name, unit_name, replace(format(warn,9),',','') warn,
-                    replace(format(crit,9),',','') crit
-                    FROM metrics AS m, index_data AS i
-                    WHERE index_id = id
-                    AND " . $l_rselector . "
-                    AND m.hidden = '0'
-                    ORDER BY m.metric_name"
+                replace(format(crit,9),',','') crit
+                FROM metrics AS m, index_data AS i
+                WHERE index_id = id
+                AND metric_id IN ( :l_rselector )
```

Comment:
Be careful, if the following condition is false 
`isset($this->metricsEnabled) && count($this->metricsEnabled) > 0`
the value of `$l_rselector` will be `index_id = '" . $this->index . "'`

Reply:
yeap for that problem i change the query
in the next commit